ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_snapshot.py

```diff
@@ -0,0 +1,357 @@
+
+    try:
+        vol_resp = client_obj.volumes.get(id=None, name=vol_name)
+        if utils.is_null_or_empty(vol_resp):
+            return (False, False, f"Volume '{vol_name}' not present on array for taking snapshot.", {})
+        snap_resp = client_obj.snapshots.get(id=None, vol_name=vol_name, name=snapshot_name)
+        if utils.is_null_or_empty(snap_resp):
+            params = utils.remove_null_args(**kwargs)
+            snap_resp = client_obj.snapshots.create(name=snapshot_name,
+                                                    vol_id=vol_resp.attrs.get("id"),
+                                                    **params)
+            if snap_resp is not None:
+                return (True, True, f"Snapshot '{snapshot_name}' created successfully.", {})
+        else:
+            return (False, False, f"Snapshot '{snapshot_name}' cannot be created as it is already present.", {})
+    except Exception as ex:
+        return (False, False, f"Snapshot creation failed | {ex}", {})
+
+
+def update_snapshot(
+        client_obj,
+        snap_resp,
+        **kwargs):
+
+    if utils.is_null_or_empty(snap_resp):
+        return (False, False, "Update snapshot failed as snapshot is not present.", {})
+
+    try:
+        snapshot_name = snap_resp.attrs.get("name")
+        changed_attrs_dict, params = utils.remove_unchanged_or_null_args(snap_resp, **kwargs)
+        if changed_attrs_dict.__len__() > 0:
+            snap_resp = client_obj.snapshots.update(id=snap_resp.attrs.get("id"), **params)
+            return (True, True, f"Snapshot '{snapshot_name}' already present. Modified the following fields:", changed_attrs_dict)
+        else:
+            return (True, False, f"Snapshot '{snapshot_name}' already present.", {})
+
+    except Exception as ex:
+        return (False, False, f"Snapshot update failed | {ex}", {})
+
+
+def delete_snapshot(
+        client_obj,
+        vol_name,
+        snapshot_name):
+
+    if utils.is_null_or_empty(snapshot_name):
+        return (False, False, "Delete snapshot failed as snapshot is not present.", {})
+    if utils.is_null_or_empty(vol_name):
+        return (False, False, "Delete snapshot failed. Volume is not present.", {})
+
+    try:
+        vol_resp = client_obj.volumes.get(id=None, name=vol_name)
+        if utils.is_null_or_empty(vol_resp):
+            return (False, False, f"Volume '{vol_name}' is not present on Array for deleting snapshot.", {})
+        snap_resp = client_obj.snapshots.get(id=None, vol_name=vol_name, name=snapshot_name)
+        if utils.is_null_or_empty(snap_resp):
+            return (False, False, f"Snapshot '{snapshot_name}' cannot be deleted as it is not present in given volume '{vol_name}'.", {})
+        else:
+            snap_resp = client_obj.snapshots.delete(id=snap_resp.attrs.get("id"))
+            # confirm
+            snap_resp = client_obj.snapshots.get(id=None, vol_name=vol_name, name=snapshot_name)
```

Comment:
Please clarify why we confirm the delete by querying the snapshot instead of trusting the result from the delete action.

Reply:
ooh.. my bad. it's not needed.

new code 
`

            client_obj.snapshots.delete(id=snap_resp.attrs.get("id"))
            return (True, True, f"Deleted snapshot '{snapshot_name}' successfully.", {})
    except Exception as ex:`